facet/README.md

```diff
@@ -0,0 +1,205 @@
+--- 
+layout: pattern
+title: Facet
+folder: facet
+permalink: /patterns/facet/
+categories:
+- structural
+language: en
+tags:
+- Decoupling
+---
+
+## Also known as
+
+Attenuation
+
+## Intent
+
+Provide an interface to a powerful object in a restricted way, either by restricting parameters 
+or only allowing calls to a subset of object's functions such that the powerful object's 
+capabilities are not abused by any other classes.
+
+## Explanation
+
+Real-world example
+
+> Imagine a knight fighting a dragon. The knight can only attempt to attack the dragon by trying
+> different attacks. Here, the facet stops the knight from directly changing the health of the 
+> dragon, as well as only allowing certain attacks to affect the dragon.
+
+In plain words
+
+> Using the facet pattern, a facet class represents strictly limited access to another class.
+
+C2 Wiki says that the intent is
+
+> Restrict an interface to obtain a smaller interface that provides less authority. Usually the 
+> smaller interface has only a subset of the methods, or allows only a subset of parameter values. 
```

Comment:
This probably needs to include more of the C2 wiki article (note that i'm not sure the article exists anymore, and so am using the wayback machine for jun 22 2021 https://web.archive.org/web/20210622051939/https://wiki.c2.com/?FacetPattern)

`Intent: Restrict an interface to obtain a smaller interface that provides less authority. Usually the smaller interface has only a subset of the methods, or allows only a subset of parameter values.
Facets are used as a security pattern in [CapabilityOrientedProgramming](https://web.archive.org/web/20210622051939/https://wiki.c2.com/?CapabilityOrientedProgramming), in order to satisfy the [PrincipleOfLeastAuthority](https://web.archive.org/web/20210622051939/https://wiki.c2.com/?PrincipleOfLeastAuthority). For example, if some client of an object only needs to be able to read information from it, that client should be provided with a read-only facet.` 

with the ordering of the two paragraphs switched may be better.